File: python/socket/socket_ephemeeral.py
#!/usr/bin/python

import sys
import getopt
import socket
import time

def main():
	host=sys.argv[1]
	port=int( sys.argv[2] )
	count=int( sys.argv[3] )

	for i in range(count):
		s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
		s.settimeout(3600)
		s.connect((host,port))
		time.sleep(1)
		s.close()
		
	

if __name__ == "__main__":
    main()

File: perl/socket/socket_ephemeeral.pl
#!/usr/bin/perl

use strict;
use warnings;
use Socket;

my $sock;
my $host=$ARGV[0];
my $port=$ARGV[1];
my $count=$ARGV[2];

my $sock_addr = sockaddr_in($port, inet_aton($host) ) or die "Cannot pack $host:$port: $!";

for(my $i=0; $i<$count; $i++){
	my $sock;
	socket($sock, PF_INET, SOCK_STREAM, getprotobyname('tcp')) or die "Cannot create socket: $!";
	connect($sock, $sock_addr) or die "Cannot connect $host:$port: $!";
	sleep(1);
}


